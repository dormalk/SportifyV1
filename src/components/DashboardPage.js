import React from 'react';

export default () => (
    <div>
        <p>wellcome my frined</p>
    </div>
);

