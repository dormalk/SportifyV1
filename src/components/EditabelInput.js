import React from 'react';
//Test
export default class EditabelInput extends React.Component{
    constructor(props){
        super();
        this.state = {
            disable: true
        }
    }

    onEditClick = () => {
        const disable = !this.state.disable;
        this.setState({disable});
    }

    render(){
        return(
            <div>
                <label><span className="input_label">{this.props.label}</span>
                    <input type="text" value={this.props.val} onChange={this.props.onChangeValue} disabled={this.state.disable}/>
                    {this.state.disable? (
                        <span className="start__edit_input" onClick={this.onEditClick}>ערוך/י</span>
                    ):(
                        <span  className="end__edit_input" onClick={this.onEditClick}>סיימ/י עריכה</span>           
                    )}
                </label>
            </div>
        )        
    }
}
