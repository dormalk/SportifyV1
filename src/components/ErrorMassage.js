import React from 'react';


export default (props) => (
    <div>
    שגיאה התרחשה - {props.error}
    </div>
) 
