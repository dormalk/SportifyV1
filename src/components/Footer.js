import React from 'react';


export default () => (
    <div className="show-for-desktop footer">
        <span>כל הזכויות שמורות</span>
    </div>
)