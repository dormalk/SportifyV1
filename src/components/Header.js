import React from 'react';
import { connect } from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({startLogout}) => (
    <div className="header">
        <div className="content-container">
            <div className="justify-container">
                <h1 className="header__title">Sportify</h1>
                <input type="text"/>
                <button onClick={startLogout}>התנתק</button>
            </div>
        </div>
    </div>
);

const mapPropsToDispatch = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});


export default connect(undefined,mapPropsToDispatch)(Header);
