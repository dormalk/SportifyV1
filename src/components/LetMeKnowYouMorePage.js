//Test
import React from 'react';
export default () =>(
    <div>
        <h1>אני רוצה להכיר אותך יותר</h1>
    </div>
)