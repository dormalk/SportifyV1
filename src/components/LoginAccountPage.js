import React from 'react';
import LoginAccountBox from './LoginAccountBox';

export const LoginAccountPage = () => (
    <div className="login-page">
        <LoginAccountBox/>
    </div>
);

export default LoginAccountPage; 

