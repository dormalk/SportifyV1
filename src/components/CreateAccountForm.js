import React from 'react';
import { connect } from 'react-redux';
import { startUploadImage } from '../actions/storage';
import { startCreateNewUser } from '../actions/user';
import ErrorMassage from './ErrorMassage';


export class CreateAccountForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            user:{
                fname: this.props.fname || '',
                lname: this.props.lname || '',
                email: this.props.email || '',
                password: this.props.password || '',
                profile: this.props.profile || ''       
            },
            c_password: '',
            error: ''
        }
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.startCreateNewUser(this.state.user).then(() => {
            this.setState(() => ({ error:'' }));
        }).catch((error) => {
            if(error.code === 'auth/email-already-in-use'){
                this.setState(() => ({ error:'המייל שהזנת נימצא בשימוש' }));
            }
            
        });
    };

    onFnameChange = (e) => {
        const fname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, fname } } ));
    }

    onLnameChange = (e) => {
        const lname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, lname } } ));
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, email } } ));
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        if(password.length < 20) {
            this.setState(() => ( { user: { ...this.state.user, password } } ));
        }
    }

    onConfirmPasswordChange = (e) => {
        const c_password = e.target.value;
        if(c_password.length < 20) {
            this.setState(() => ( { c_password } ));
        }
    }

    checkCompatibilityPasswords = (e) => {
        const r_password = this.state.user.password;
        const c_password = this.state.c_password;
        if(c_password !== r_password){
            this.setState(() => ({ error:'Passwords do not match' }));
        }
        else{
            this.setState(() => ({ error:'' }));
        }
    }

    onProfileChange = (e) => {
        const picture = e.target.files[0];
        if(picture.type !== 'image/jpeg' && picture.type !== 'image/jpg' && picture.type !== 'image/png'){
            this.setState({error:'נא בחר קובץ תמונה (jpg,jpeg,png)'});
        }
        else if(picture.size > 585750){
            this.setState({error:'נא בחר קובץ תמונה פחות מ 585,750 בית'});
            console.log('Hello');
        }
        else{
                this.setState(() => ({error:''}));  
                this.props.startUploadImage(picture).then((profile) => {
                    this.setState({error:''});
                    this.setState(() => ( { user: { ...this.state.user, profile } } ));
                }).catch((error) => {
                    this.setState({error:'לא ניתן לעלות קובץ תמונה ברגע זה'});
                });
        }
    }

    render(){
        return (
            <div>
                {this.state.error && <ErrorMassage error={this.state.error}/>}
                <div className="content-container">
                    <div className="signup-box">
                        <form 
                            onSubmit={this.onSubmit}
                            className="signup-form"
                        >
                            <h2>יצירת משתמש חדש</h2>
                            {this.state.error && <p className="form__error">{this.state.error}</p>}
                            <label>שם פרטי
                                <input 
                                    type="text"
                                    onChange={this.onFnameChange}
                                    required
                                />
                            </label>
                            <label>שם משפחה
                                <input 
                                    type="text"
                                    onChange={this.onLnameChange}
                                    required
                                />
                            </label>
                            <label>אימייל
                                <input 
                                    type="text"
                                    onChange={this.onEmailChange}
                                    required
                                />
                            </label>
                            <label>סיסמא
                                <input 
                                    type="password"
                                    onChange={this.onPasswordChange}
                                    required
                                />
                            </label>
                            <label>אימות סיסמא
                                <input 
                                    type="password"
                                    onChange={this.onConfirmPasswordChange}
                                    onBlur={this.checkCompatibilityPasswords}
                                    required
                                />
                            </label>
                            <label>תמונת פרופיל
                                <input type="file" onChange={this.onProfileChange}/>
                            </label>
                            {this.state.profile && <img src={this.state.profile}/>}
                            <button className="signup__box">צור חשבון</button>
                        </form>
                    </div>
                </div>
            </div>
        );
    };
};

const mapPropsToDispatch = (dispatch) => ({
    startUploadImage: (picture) => dispatch(startUploadImage(picture)),
    startCreateNewUser: (user) => dispatch(startCreateNewUser(user))
});


export default connect(undefined,mapPropsToDispatch)(CreateAccountForm);