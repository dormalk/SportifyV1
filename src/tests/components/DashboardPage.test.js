import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPage } from '../../components/DashboardPage';

test('should render DashboardPage correctly', () => {
    const wrapper = shallow(<DashboardPage/>);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<DashboardPage startLogout={startLogout}/>);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});