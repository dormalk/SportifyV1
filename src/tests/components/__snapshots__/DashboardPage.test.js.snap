// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
<div>
  <p>
    wellcome my frined
  </p>
</div>
`;
